subworkflows/nf-core/fastqc_umitools_trimgalore.nf

```diff
@@ -6,6 +6,23 @@ include { FASTQC           } from '../../modules/nf-core/modules/fastqc/main'
 include { UMITOOLS_EXTRACT } from '../../modules/nf-core/modules/umitools/extract/main'
 include { TRIMGALORE       } from '../../modules/nf-core/modules/trimgalore/main'
 
+//
+// Function that parses TrimGalore log output file to get total number of reads after trimming
+//
+def getTrimGaloreReadsAfterFiltering(log_file) {
+    def total_reads = 0
+    def filtered_reads = 0
+    log_file.eachLine { line ->
+        def total_reads_matcher = line =~ /([\d\.]+)\ssequences processed in total/
+        def se_filtered_reads_matcher = line =~ /shorter than the length cutoff of\s[\d\.]+\sbp:\s([\d\.]+)/
```

Comment:
I think with 

`
def filtered_reads_matcher = line =~ /shorter than the length cutoff[^:]+:\s([\d\.]+)/`

you could match both, single and paired end reports simultaneously. That regex at least works on all trimming reports that are either contained in the [Test data of TrimGalore](https://github.com/FelixKrueger/TrimGalore/tree/master/test_files) as well as [MultiQC Test data](https://github.com/ewels/MultiQC_TestData/tree/master/data/ngi/data/trim_galore).

Reply:
Thanks! Nice regex fu! I will test this out and add it in a follow up PR to save re-running the tests here👍🏽